### ISSUE TITLE
Pushing template assets doesn't work and they are not used in 2.x, remove this feature

### ISSUE THREAD

**Issue by HollyPony** [punkave/apostrophe#1103]
Since this function taking `req, asset` params is called with asset as parameter only.

Function definition : https://github.com/punkave/apostrophe/blob/4b9039cdf38ef0a7ffed9b30e9d4991d61bf22c5/lib/modules/apostrophe-assets/index.js#L1173

Function call: https://github.com/punkave/apostrophe/blob/4b9039cdf38ef0a7ffed9b30e9d4991d61bf22c5/lib/modules/apostrophe-assets/index.js#L1463

It may be require a test case.

I found this bug trying to make this:

```javascript
// lib/modules/apostrophe-assets/index.js
  afterConstruct: (self) => {
    self.pushAsset('template', (req, data) => {}, {
      when: 'always',
      data: self.apos.templates.safe('<div>Hello</div>')
    })
  }
```

Describe here : http://apostrophecms.org/docs/modules/apostrophe-assets/index.html#push

EDIT: If I remove the `req` from function def (seems unused),  I get this Error : 
`Error: render() must not be called from a Nunjucks helper function nested inside another call to render(). Use partial() instead.`

**Comment by boutell** [comment]
Hi Arthur,

We haven't used the template push mechanism once in 2.x! We moved away from it because it cluttered up the DOM with a lot of templates that might not get used on a given page request. Instead we have consistently made AJAX requests (usually with the .html method of apostrophe-context, which apostrophe-modal inherits from...) any time we wanted the server to render something that isn't part of the initial page.

(We did use this mechanism  a lot in 0.5.)

Since it apparently has been broken all along in 2.x, removing it would not be a bc break, so we might just do that. However I will take a closer look when I get a moment.
